sinker/backend/pktvisor/pktvisor.go

```diff
@@ -84,103 +85,117 @@ func parseToProm(ctxt *context, stats StatSnapshot) prometheus.TSList {
 	return tsList
 }
 
-func convertToPromParticle(ctxt *context, m map[string]interface{}, label string, tsList *prometheus.TSList) {
-	for k, v := range m {
-		switch c := v.(type) {
+func convertToPromParticle(ctxt *context, statsMap map[string]interface{}, label string, tsList *prometheus.TSList) {
+	for key, value := range statsMap {
+		switch statistic := value.(type) {
 		case map[string]interface{}:
 			// Call convertToPromParticle recursively until the last interface of the StatSnapshot struct
-			// The prom particle label it's been formed during the recursive call
-			convertToPromParticle(ctxt, c, label+k, tsList)
+			// The prom particle label it's been formed during the recursive call (concatenation)
+			convertToPromParticle(ctxt, statistic, label+key, tsList)
 		// The StatSnapshot has two ways to record metrics (i.e. Live int64 `mapstructure:"live"`)
 		// It's why we check if the type is int64
 		case int64:
 			{
 				// Use this regex to identify if the value it's a quantile
 				var matchFirstQuantile = regexp.MustCompile("^([P-p])+[0-9]")
-				if ok := matchFirstQuantile.MatchString(k); ok {
+				if ok := matchFirstQuantile.MatchString(key); ok {
 					// If it's quantile, needs to be parsed to prom quantile format
-					tsList = makePromParticle(ctxt, label, k, v, tsList, ok)
+					tsList = makePromParticle(ctxt, label, key, value, tsList, ok, "")
 				} else {
-					tsList = makePromParticle(ctxt, label+k, "", v, tsList, false)
+					tsList = makePromParticle(ctxt, label+key, "", value, tsList, false, "")
 				}
 			}
 		// The StatSnapshot has two ways to record metrics (i.e. TopIpv4   []NameCount   `mapstructure:"top_ipv4"`)
 		// It's why we check if the type is []interface
 		// Here we extract the value for Name and Estimate
 		case []interface{}:
 			{
-				for _, value := range c {
+				for _, value := range statistic {
 					m, ok := value.(map[string]interface{})
 					if !ok {
 						return
 					}
-					var lbl string
-					var dtpt interface{}
+					var promLabel string
+					var promDataPoint interface{}
 					for k, v := range m {
 						switch k {
 						case "Name":
 							{
-								lbl = fmt.Sprintf("%v", v)
+								promLabel = fmt.Sprintf("%v", v)
 							}
 						case "Estimate":
 							{
-								dtpt = v
+								promDataPoint = v
 							}
 						}
 					}
-					tsList = makePromParticle(ctxt, label+k, lbl, dtpt, tsList, false)
+					tsList = makePromParticle(ctxt, label+key, promLabel, promDataPoint, tsList, false, key)
 				}
 			}
 		}
 	}
 }
 
-func makePromParticle(ctxt *context, label string, k string, v interface{}, tsList *prometheus.TSList, quantile bool) *prometheus.TSList {
+func makePromParticle(ctxt *context, label string, k string, v interface{}, tsList *prometheus.TSList, quantile bool, name string) *prometheus.TSList {
 	mapQuantiles := make(map[string]float64)
-	mapQuantiles["P50"] = 0.50
-	mapQuantiles["P90"] = 0.90
+	mapQuantiles["P50"] = 0.5
```

Comment:
did you test that this fixes the problem? i'm worried that it needs to be a string, not a number